lib/apple/index.js

```diff
@@ -37,13 +37,30 @@ function parseResult(result) {
 	return {
 		receipt: receipt,
 		/* jshint camelcase:false */
-		transactionId: receipt.transaction_id,
-		productId: receipt.product_id
+		transactionId: getReceiptFieldValue(receipt, "transaction_id"),
+		productId: getReceiptFieldValue(receipt, "product_id")
 		/* jshint camelcase:true */
 	};
 }
 
 
+function getReceiptFieldValue(receipt, field)
+{
+    var fieldValue = null;
+
+    if (receipt.hasOwnProperty(field))
+    {
+        fieldValue = receipt[field];
+    }
+    else if (receipt.hasOwnProperty("in_app") && receipt.in_app[0].hasOwnProperty(field))
```

Comment:
For that way of referring to in_app however, you will have to add `/* jshint camelcase:false */` in this function (see above)